python/vulnerabilities/count_unique_cves.py

```diff
@@ -0,0 +1,323 @@
+    wait_minutes = 20
+    interval_secs = 5
+    wait_count = round(wait_minutes * (60 / interval_secs))
+    cnt = 1
+    ready = False
+
+    # Check the export status until the export is ready or the time limit is met.
+    while not ready and cnt < wait_count:
+        print(f"Sleeping for {interval_secs} seconds.  ({cnt} out of {wait_count})\r", end='')
+        time.sleep(interval_secs)
+        cnt += 1
+
+        ready = get_export_status(base_url, headers, search_id)
+
+    print("")
+    if cnt >= wait_count:
+        print_info(f"Waited for {wait_minutes} minutes.")
+        print_info(f"Consider re-running with search ID")
+        sys.exit(1)
+ 
+# Obtain the exported vuln data and gunzip it.
+def retrieve_vuln_data(base_url, base_headers, id, vuln_file_name):
+    jsonl_vuln_file_name = f"{vuln_file_name}.jsonl"
+    
+    # Check if the JSONL file already exists.  This is useful when developing 
+    # new process code after the export.
+    if os.path.exists(jsonl_vuln_file_name):
+        print_info(f"{jsonl_vuln_file_name} already exists, so we're using it.")
+        return jsonl_vuln_file_name
+
+    get_data_url = f"{base_url}/data_exports/?search_id={id}"
+    headers = base_headers.copy()
+    headers['Accept'] = "application/gzip; charset=utf-8"
+    
+    gz_vuln_file_name = vuln_file_name + ".gz"
+
+    # Retrieve the gzip file by streaing.
+    response = requests.get(get_data_url, headers=headers, stream=True)
+    if response.status_code != 200:
+        process_http_error(f"Retrieve Data Export API Error", response, get_data_url)
+        sys.exit(1)
+
+    try:
+        with open(gz_vuln_file_name, 'wb') as file_gz:
+            for block in response.iter_content(8192):
+                file_gz.write(block)
+    
+    except Exception as exp:
+        print_error(f"Retrieve vuln data Error: {exp.__str__()}")
+        sys.exit(1)
+
+    # Gunzip the file into the JSONL file.
+    print_info(f"Gunzipping file {gz_vuln_file_name} to {jsonl_vuln_file_name}")
+    with gzip.open(gz_vuln_file_name, 'rb') as f_in:
+        with open(jsonl_vuln_file_name, 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+
+    print_info(f"File {gz_vuln_file_name} gunzipped to {jsonl_vuln_file_name}")
+    f_out.close()
+    f_in.close()
+    return jsonl_vuln_file_name
+
+# Count the number of lines in the unzip asset file.
+# From: https://stackoverflow.com/questions/845058/how-to-get-line-count-of-a-large-file-cheaply-in-python
+def count_lines(vuln_file_name):
+    print_info(f"Counting lines in {vuln_file_name}")
+
+    f = open(vuln_file_name, 'rb')
+    lines = 0
+    buf_size = 1024 * 1024
+    read_f = f.raw.read
+
+    buf = read_f(buf_size)
+    while buf:
+        lines += buf.count(b'\n')
+        buf = read_f(buf_size)
+    
+    f.close()
+
+    # Add one to the number of lines because the end of files does not have '\n'.
+    return lines + 1
+
+def convert_to_json(vuln_line):
```

Comment:
can we not directly use json format ?

Reply:
Glad you caught this.  This is really convert_from_jsonl().